backend-config/backend-config.go

```diff
@@ -407,3 +410,167 @@ func (bc *backendConfigImpl) Identity() identity.Identifier {
+	// Only when 1 & 2 are true, we would allow the event to flow through to server
+	evaluatedDefaultBehaviour := isSupportedMsgType && destination.IsProcessorEnabled
+
+	/*
+		In future(near one) this will not be a string but rather an object
+		For every connection made to destination, we would basically have
+		```
+		{
+			connectionMode: {
+				[srcType]: oneof[cloud, hybrid, device]
+			}
+		}
+
+		```
+		Example
+		{
+			...
+			"connectionMode": {
+				"web": "", // one of cloud, hybrid or device
+				"cordova": "", // one of cloud, hybrid or device
+				"python": "", // one of cloud, hybrid or device
+				"cloud": "" // one of cloud, hybrid or device
+			}
+			...
+		}
+	*/
+	// New logic for evaluating "connectionMode"
+	var destConnModeI interface{}
+	for _, val := range []string{srcType, strings.ToLower(source.SourceDefinition.Name)} {
+		destConnModeI = misc.MapLookup(destination.Config, "connectionMode", val)
+		if destConnModeI != nil {
+			break
+		}
+	}
+	// destConnModeI := misc.MapLookup(destination.Config, "connectionMode")
+	if destConnModeI == nil {
+		return evaluatedDefaultBehaviour
+	}
+	destConnectionMode, isDestConnModeMapOfString := destConnModeI.(string)
+	if !isDestConnModeMapOfString {
+		pkgLogger.Errorf("Given destination connection mode :%v, cannot be casted to string", destConnModeI)
+		return false
+	}
+
+	supportedConnectionModesI, connModesOk := destination.DestinationDefinition.Config["supportedConnectionModes"]
+	if !connModesOk {
+		// Probably the "supportedConnectionModes" key is not present, so we rely on Default behaviour
+		return evaluatedDefaultBehaviour
+	}
+	supportedConnectionModes := supportedConnectionModesI.(map[string]interface{})
+
+	supportedEventPropsMapI := misc.MapLookup(supportedConnectionModes, destConnectionMode, srcType)
+	if supportedEventPropsMapI == nil {
+		// Most probably sourceType would not be present in "supportedConnectionModes"
+		// Might also occur when destination connection mode is not provided in "supportedConnectionModes"
+		return evaluatedDefaultBehaviour
+	}
+	supportedEventPropsMap := supportedEventPropsMapI.(map[string]interface{})
+	// Flag indicating to let the event pass through
+	allowEvent := evaluatedDefaultBehaviour
+	for eventProperty, supportedEventVals := range supportedEventPropsMap {
+		if !allowEvent {
+			allowEvent = evaluatedDefaultBehaviour
+			break
+		}
+		switch eventProperty {
+		case "messageType":
+			supportedVals := ConvertToArrayOfType[string](supportedEventVals)
+			// TODO: Should remove this
+			pkgLogger.Infof("SupportedVals: %v -- EventType from event: %v\n", supportedVals, eventType)
+			allowEvent = lo.Contains(supportedVals, eventType) && evaluatedDefaultBehaviour
+		}
+	}
+	return allowEvent
+}
+
+func evaluateSupportedTypes[T comparable](destConfig map[string]interface{}, evalKey string, checkValue T) bool {
+	if !lo.Contains([]string{"supportedMessageTypes"}, evalKey) {
```

Comment:
required?
evalKey !="supportedMessageTypes"

Reply:
Initially we had `supportedSourceTypes` as well here. So the condition would look something like below
```
if !lo.Contains([]string{"supportedMessageTypes", "supportedSourceTypes"}, evalKey)
```

since `supportedSourceTypes` didn't make a lot of sense, removed it.
Hence the logic looks like this.

Reply:
I have simplified the condition for now!